draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -270,262 +296,371 @@ of testing):
+-->
+
 We assume that the values produced and consumed by the above functions are all
 byte strings, with fixed lengths:
 
-- `Nseed`: The length of a key seed (input to DeriveKeyPair)
-- `Nek`: The length of a public encapsulation key
-- `Ndk`: The length of a secret decapsulation key
-- `Nct`: The length of a ciphertext produced by Encaps
-- `Nss`: The length of a shared secret produced by Encaps or Decaps
+- `Nseed`: The length in bytes of a key seed (input to DeriveKeyPair)
+- `Nek`: The length in bytes of a public encapsulation key
+- `Ndk`: The length in bytes of a secret decapsulation key
+- `Nct`: The length in bytes of a ciphertext produced by Encaps
+- `Nss`: The length in bytes of a shared secret produced by Encaps or Decaps
 
-This interface is effectively the same as the one defined in the Hybrid Public
-Key Encryption (HPKE) specification {{?RFC9180}}.  The only difference is that
-here we assume that all values are byte strings, whereas in HPKE keys are opaque
-by default and serialized or deserialized as needed.  We also use slightly
-different terminology for keys, emphasizing "encapsulation" and "decapsulation"
-as opposed to "public" and "secret".
+> This interface is effectively the same as the one defined in the Hybrid Public
+> Key Encryption (HPKE) specification {{?RFC9180}}.  The only difference is that
+> here we assume that all values are byte strings, whereas in HPKE keys are
+> opaque by default and serialized or deserialized as needed.  We also use
+> slightly different terminology for keys, emphasizing "encapsulation" and
+> "decapsulation" as opposed to "public" and "secret".
 
-## `XOF` {#xof}
+## Hash functions {#hash}
 
-Extendable-output function (XOF). A function on bit strings in which the
-output can be extended to any desired length. Ought to satisfy the following
-properties as long as the specified output length is sufficiently long to
-prevent trivial attacks:
+Functionally, a hash function is simply a function that produces a fixed-length
+output byte string from an input byte string of arbitrary length.
+
+- `Nh` - The length in bytes of an output from this hash function.
+- `Hash(input) -> output`: Produce a byte string of length `Nh` from an input
+  byte string.
 
-1. (One-way) It is computationally infeasible to find any input that maps to
-   any new pre-specified output.
+For simplicity, we will write invocations of a hash function without `Hash`
+being explicit.  Invoking a hash function `Foo` on input `input` will be written
+as `Foo(input)` instead of `Foo.Hash(input)`.
+
+Hash function used with the constructions in this document should be structured
+so that they can be regarded as random oracles in either the classical or
+quantum random oracle model.  (Note that this property implies standard hash
+function properties such as collision resistance and second preimage
+resistance.)  Each hash function we refer to should be an independent random
+oracle.
+
```

Comment:
All mention of 'KDF' and 'XOF' has been removed and 'hash functions' has been added, these things are not the same